app/modules/intelligence/agents/chat_agents/debugging_chat_agent.py

```
@@ -47,16 +50,23 @@ def __init__(self, mini_llm, reasoning_llm, db: Session):
         self.db = db
 
     @lru_cache(maxsize=2)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)